rfcs/20200612-stream-executor-c-api.md

```diff
@@ -148,38 +206,37 @@ typedef struct SP_AllocatorStats {
   int64_t largest_free_block_bytes;
 } SP_AllocatorStats;
 
-#define SP_ALLOCATORSTATS_STRUCT_SIZE TF_OFFSET_OF_END(SP_AllocatorStats, largest_free_block_bytes)
+#define SP_ALLOCATORSTATS_STRUCT_SIZE \
+  TF_OFFSET_OF_END(SP_AllocatorStats, largest_free_block_bytes)
 
+// Potential states for an SP_Event. If `poll_for_status` returns anything aside
+// from kPending or kComplete, an error has occurred; kUnknown is a bad state.
 typedef enum SE_EventStatus {
   SE_EVENT_UNKNOWN,
   SE_EVENT_ERROR,
   SE_EVENT_PENDING,
   SE_EVENT_COMPLETE,
 } SE_EventStatus;
 
-typedef struct SE_Options {
-  size_t struct_size;
-  void* ext;
-  int32_t ordinal;
-} SE_Options;
-
-#define SE_OPTIONS_STRUCT_SIZE TF_OFFSET_OF_END(SE_Options, ordinal)
-
-typedef struct SE_DeviceMemoryBase {
+// Memory allocation information.
+// This matches DeviceMemoryBase defined here:
+// https://cs.opensource.google/tensorflow/tensorflow/+/refs/tags/v2.3.0:tensorflow/stream_executor/device_memory.h;l=57
+typedef struct SP_DeviceMemoryBase {
   size_t struct_size;
-  void* ext;
+  void* ext;  // free-form data set by plugin
+  // Platform-dependent value representing allocated memory.
   void* opaque;
-  uint64_t size;
-  uint64_t payload;
-} SE_DeviceMemoryBase;
+  uint64_t size;     // Size in bytes of this allocation.
+  uint64_t payload;  // Value for plugin's use
+} SP_DeviceMemoryBase;
 
-#define SE_DEVICE_MEMORY_BASE_STRUCT_SIZE TF_OFFSET_OF_END(SE_DeviceMemoryBase, payload)
+#define SP_DEVICE_MEMORY_BASE_STRUCT_SIZE \
+  TF_OFFSET_OF_END(SP_DeviceMemoryBase, size)
```

Comment:
```suggestion
  TF_OFFSET_OF_END(SP_DeviceMemoryBase, payload)
```

(Please check the original .h to see that the bug isn't there as well.)

Reply:
@annarev just fixed it in https://github.com/tensorflow/tensorflow/commit/c533255de41b0155e3517950126af43a6f8eb233. So fast!